packages/strapi-plugin-graphql/services/Mutation.js

```diff
@@ -195,27 +195,49 @@ module.exports = {
 
       // Resolver can be a function. Be also a native resolver or a controller's action.
       if (_.isFunction(resolver)) {
-        context.params = Query.convertToParams(
-          options.input.where || {},
-          (plugin ? strapi.plugins[plugin].models[name] : strapi.models[name])
-            .primaryKey
-        );
-        context.request.body = options.input.data || {};
+        const normalizedName = _.toLower(name);
+
+        let primaryKey;
+
+        if (plugin) {
+          primaryKey = strapi.plugins[plugin].models[normalizedName].primaryKey;
+        } else {
+          primaryKey = strapi.models[normalizedName].primaryKey;
+        }
+
+        if (options.input && options.input.where) {
+          context.params = Query.convertToParams(
+            options.input.where || {},
+            primaryKey
+          );
+        } else {
+          context.params = {};
+        }
+
+        if (options.input && options.input.data) {
+          context.request.body = options.input.data || {};
+        } else {
+          context.request.body = {};
```

Comment:
shouldn't we just pass all the options as body in case the user declare a mutation with custom arguments ?

Reply:
Do you mean
`context.request.body = options.input.data;`
Instead of
`context.request.body = options.input.data || {};`
?

Reply:
nop I mean 
```js
if (options.input && options.input.data) {
     context.request.body = options.input.data || {};
} else {
     context.request.body = options; // here you can pass the grahpql arguments as body to the controller
}
```

Reply:
Ah yeah. Will do that.